### PR TITLE
feat: add index link to each section of rule group

### LINKED ISSUE #940: feat: Add link to index page for each AIP in the linter documentation
## User Story

**As an [`api-linter` docs](https://linter.aip.dev) reader I should be able to link to the index page for linter documentation for a specific AIP so that I can see all of the aspects of the AIP that the linter covers for that AIP in one place.**

## Problem

For each rule set, e.g. [core](https://linter.aip.dev/rules/core/), the AIP link expands the folio but there is no link available to go to index.md for that the specific AIP for which that documentation exists.

## Acceptance Criteria

1. Given that I  visit an `aip-linter` rules group page, e.g. [core](https://linter.aip.dev/rules/core/), there is a link for each AIP
2. And if I click that link corresponding to an AIP
3. Then I am directed to the index page for that the `api-linter` documentation for that AIP

### PR DESCRIPTION
**Pull request by shwoodard**
In the docs, on the listing for the rule group, e.g. core, add an
[index] link at the top of each accordion section that links to the
index page of the api linter documention for that aip.

`[index]` link here,

image

links to 

image

Closes #940

### CODE REVIEW

File: docs/_includes/linter-group-listing.html [hunk 1]
```diff
@@ -36,6 +36,11 @@ <h4 class="glue-expansion-panel__button-header">
     <div class="glue-expansion-panel-content" id="panel-aip-{{ pbr.name }}">
       <div class="has-rule-listing">
         <table class="glue-table--datatable glue-table--stacked api-linter-rule-listing" style="width: 100%;">
+          <tr>
+            <td>
+              <a href="/{{ pbr.name }}">[index]</a>
```
Comment (noahdietz) [inline docs/_includes/linter-group-listing.html:41]: Should this be `p` since the loop is `for p in pbr.items`? I honestly don't know.

**Review by noahdietz — COMMENTED** [review]
Would this allow me to link to a ruleset like so: `https://linter.aip.dev/rules/core/122`?

File: docs/_includes/linter-group-listing.html (hunk 1 above)
Comment (shwoodard) [reply docs/_includes/linter-group-listing.html:41]: We want the outer loop var, `pbr`, because this is the link to the index of all `p`.

**Comment by shwoodard** [comment]
> Would this allow me to link to a ruleset like so: `https://linter.aip.dev/rules/core/122`?

No, it doesn't change any routing, it just ads a link so that the accordion section has an `[index]` link that links to the linter AIP index docs page, like https://linter.aip.dev/122. I can add that feature though.

**Comment by noahdietz** [comment]
like https://linter.aip.dev/122. I can add that feature though.

Oh no need, that's perfect. Thanks!

**Review by noahdietz — APPROVED** [review]

**Pull request merged**